lib/actions/appleMusic.js

```diff
@@ -155,15 +145,16 @@ function appleMusic(player, values) {
                 return player.coordinator.clearQueue()
                   .then(() => player.coordinator.addURIToQueue(queueTracks[0].uri, queueTracks[0].metadata, true, 1))
                   .then(() => {
-                      if (radioOn(player)) {
-                        return promise.then(() => player.coordinator.setAVTransport(queueURI,''));
+                      if (isRadioOrLineIn(player.coordinator.state.currentTrack.uri)) {
+                        player.coordinator.setAVTransport(queueURI,''));
                       }
+                      return Promise.resolve();
```

Comment:
You don't have to explicitly return a Promise within a promise chain, minor cosmetic thing. Also, if you are starting a radio, you probably want to skip the track queueing step, so maybe you should have a conditional here instead and return either setAVtransport with radio URI, or the promise that queues all tracks (and final setAVTransport for queue playback).